File: python/programmersLV2/기능개발.py
# 1st hit
from collections import deque

def solution(progresses, speeds):
    progresses = deque()
    speeds = deque()
    release = 0
    answer = []
    while True:
        if len(progresses) == 0:
            break
        for i in range(0, len(progresses)):
            progresses[i] += speeds[i]
        while True:
            if progresses[0] >= 100:
                progresses[0].popleft()
                speeds[0].popleft()
                release +=1
            else:
                answer.append(release)
                release = 0
                break

    return answer

print(solution([93, 30, 55], [1, 30, 5]))

File: algorithm/src/programmers/기능 개발.kt
package programmers

import java.util.*

class Solution_5 {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var day = 0
        var cnt = 0
        var answer = intArrayOf()

        var progressQueue : Queue<Int> = LinkedList<Int>()
        var speedsQueue : Queue<Int> = LinkedList<Int>()

        for (i in 0 until progresses.size) {
            progressQueue.add(progresses[i])
            speedsQueue.add(speeds[i])
        }

        while(!progressQueue.isEmpty()) {
            if(progressQueue.element() + speedsQueue.element() * day >= 100) {
                progressQueue.remove()
                speedsQueue.remove()
                cnt++
            }
            else {
                if(cnt >= 1) {
                    answer = answer.plus(cnt)
                    cnt = 0
                }
                day++
            }
        }
        answer = answer.plus(cnt)

        return answer

    }
}